File: backend/src/Core/Dev4All.Domain/Entities/ContractRevision.cs
using Dev4All.Domain.Common;
using Dev4All.Domain.Exceptions;

namespace Dev4All.Domain.Entities;

public class ContractRevision : BaseEntity
{
    public Guid ContractId { get; private set; }
    public string RevisedById { get; private set; } = string.Empty;
    public string ContentSnapshot { get; private set; } = string.Empty;
    public int RevisionNumber { get; private set; }
    public string? RevisionNote { get; private set; }

    public Contract Contract { get; set; } = null!;

    public static ContractRevision CreateSnapshot(
        Guid contractId, string revisedById, string contentSnapshot, int revisionNumber, string? note = null)
    {
        if (contractId == Guid.Empty)
            throw new BusinessRuleViolationException("Contract ID cannot be empty.");

        return new ContractRevision
        {
            ContractId = contractId,
            RevisedById = revisedById,
            ContentSnapshot = contentSnapshot,
            RevisionNumber = revisionNumber,
            RevisionNote = note
        };
    }
}

File: mobile/core/domain/src/main/kotlin/com/dev4all/mobile/core/domain/model/ContractRevision.kt
package com.dev4all.mobile.core.domain.model

/**
 * Domain model for a contract revision snapshot.
 * Field names match backend Dev4All.Domain.Entities.ContractRevision.
 */
data class ContractRevision(
    val id: String,
    val contractId: String,
    val revisedById: String,
    val contentSnapshot: String,
    val revisionNumber: Int,
    val revisionNote: String? = null,
    val createdDate: String = "",
)
